app/src/main/java/org/mozilla/fenix/StrictModeManager.kt

```diff
@@ -47,26 +44,22 @@ object StrictModeManager {
                 builder.detectContentUriWithoutPermission()
             }
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                if (setPenaltyDeath || setPenaltyDialog) {
-                    builder.permitNonSdkApiUsage()
-                } else {
-                    builder.detectNonSdkApiUsage()
-                }
+                builder.detectNonSdkApiUsage()
```

Comment:
So, I'm not sure why we added this exactly. But I remember Sachin added this check because Firebase kept screwing up with random StrictMode violations with the `detectNonSdkApiUsage`. Do you think we should let this one be and if Firebase has issues we'll change it then?

Reply:
I mention this in the extended commit summary for the commit that made this change:

> We don't use penalty death for the VM policy so we theoretically don't
need to disable this check if penalty death is enabled.

Since there is no comment explaining why this theory doesn't make sense, we can't know whether this is true or not without looking at the issue https://github.com/mozilla-mobile/fenix/issues/7225 and PR https://github.com/mozilla-mobile/fenix/pull/10831 – but I don't see any info there – or trying to change it. Given we have no information, I'd rather try to clean up the code than leaving around "if we touch it, it may break!" type issues, gradually increasing the complexity of the code (e.g. remember that bug we fixed where we unnecessarily `setAdapter(null); setAdapter(content)` on every data refresh? Imagine if someone cleaned that up for us already because they didn't know what it was and figured it was unnecessary).

That being said, it's about cleaning up gradually – we're making a handful of changes here so we could make one fewer and clean this up later. It's up to you.

Reply:
I agree with that reasoning. The `setAdapter` is a good example. Since we have context now, it might be better to get it out since it seems to not affect our code. I vote to get it removed since it did work locally on an emulated pixel 2 with Android P, so this works for me.